### ISSUE TITLE
Flatpack file system protection may be source of errors for uploads/downloads.

### ISSUE THREAD

**Issue by mova** [getferdi/ferdi#1070]
TLDR: Fix flatpack permissions pls

Hi there!
I noticed that couldn't use Ferdi to share photos on Whatsapp&Telegram anymore after reinstalling my system.
Additionally the upload menu only shows parts of the filesystem. 
Ferdi installed via flatpack is run inside  a ``bwrap`` process that restricts the access to the filesystem:

```
>flatpak info --show-permissions com.getferdi.Ferdi 
[Context]
shared=network;ipc;
sockets=x11;pulseaudio;
devices=all;
filesystems=xdg-download;xdg-pictures:ro;xdg-videos:ro;xdg-documents:ro;/run/.heim_org.h5l.kcm-socket;

[Session Bus Policy]
org.kde.StatusNotifierWatcher=talk
org.freedesktop.Notifications=talk
org.kde.*=own
```
This only allows access to some folders.
Sidenote: Additionally this setting may deny ferdi access to gpu acceleration (?).

I tried ``sudo flatpak override --filesystem=home com.getferdi.Ferdi`` to give ferdi access to my home system, but I could override it. I ended up installing flatseal and granting the permission in the gui, which fixed to problem (after killing all ferdi processes).

I found some bug reports that may be caused by this:

https://github.com/getferdi/ferdi/issues/1064
https://github.com/getferdi/ferdi/issues/965
https://github.com/getferdi/ferdi/issues/751
https://github.com/getferdi/ferdi/issues/743
https://github.com/getferdi/ferdi/issues/523
https://github.com/getferdi/ferdi/issues/469

**Comment by vantezzen** [comment]
Thank you for investigating this issue!
I haven't worked with Flatpak yet so I can't really look into this. @lhw originally published Ferdi on Flatpak - @lhw can help with fixing this?

**Comment by lhw** [comment]
Yes. That is definitely a issue that should be in https://github.com/flathub/com.getferdi.Ferdi/issues not here. But allowing full access to the home directory would invalidate all benefits gained from limiting the filesystem access to the sandbox for most if not all PCs. And I don't really see ferdi being in use on servers.
The `devices=all` line should allow access to opengl rendering for Ferdi. So that part should be fine.

All in all I don't see any benefit in allowing these permissions when an option such as flatseal exists that would allow a user to give more permissions on demand. It would be great if flatpak could provide a dialogue to make it easier for users to change permissions but I think the defaults should be on the safe side.

**Comment by mova** [comment]
I think most users are going to need access to the file system anyway, so I'm not sure of the benefit.
In my opinion you are sacrificing a lot of comfort for very little security.
If we could integrate granting this permission in Ferdi, that would be fine but, the current state is not acceptable from a user perspective. 
You just don't expect that you need to set a setting in an additional external tool to share images in Telegram or download an 
attachment from your Gmail, as the numerous bug reports are showing.

**Comment by lhw** [comment]
Alright. I'll think about it. I don't think we can transfer the issue to the flathub project. Just open a new one over there and I'll get around to it sometime in the next few days.

**Comment by sbglasius** [comment]
@mova's suggestion was almost right

This gives access to the full `$USER_HOME`:
```
sudo flatpak override --filesystem=$HOME com.getferdi.Ferdi
``` 
While it's probably not what flatpak developers intended, it works for me.

**Comment by lhw** [comment]
The flatpak is now updated to include the filesystem access change. This issue can be closed as well I assume.
